pkg/codegen/dotnet/gen.go

```diff
@@ -242,6 +246,12 @@ func (mod *modContext) typeString(t schema.Type, qualifier string, input, state,
 		case mod.details(t).functionType:
 			typ += "Result"
 		}
+	case *schema.ResourceType:
+		typ = mod.tokenToResource(t.Token)
+		if typ != "" {
+			typ += "."
+		}
+		typ += tokenToName(t.Token)
```

Comment:
Any reason these extra lines aren't inside `tokenToResource`'s implementation? Or just not have `tokenToResource` and use `mod.tokenToNamespace(t.Token, "")` on line 250?